src/thor/triplegbuilder.cc

```diff
@@ -766,6 +808,10 @@ TripLegBuilder::Build(const AttributesController& controller,
   const DirectedEdge* prev_de = nullptr;
   // TODO: this is temp until we use transit stop type from transitland
   TransitPlatformInfo_Type prev_transit_node_type = TransitPlatformInfo_Type_kStop;
+
+  if (controller.category_attribute_enabled(kShapeAttributesCategory)) {
```

Comment:
instead of having this `if` twice you could move it to the beginning of the function before the special case single edge route